protocols/protocols.go

```
@@ -66,7 +68,17 @@ func MapTCPProtocolHandlers(log interfaces.Logger, h interfaces.Honeypot) map[st
 		return tcp.HandleADB(ctx, conn, md, log, h)
 	}
 	protocolHandlers["tcp"] = func(ctx context.Context, conn net.Conn, md connection.Metadata) error {
+		conn.SetReadDeadline(time.Now().Add(500 * time.Millisecond))
 		snip, bufConn, err := Peek(conn, 4)
+		var netErr net.Error
+		if errors.As(err, &netErr) && netErr.Timeout() {
+			if err := tcp.SendBanner(md.TargetPort, conn, md, log, h); err != nil {
+				log.Error("Failed to send service banner", producer.ErrAttr(err))
+			}
+			conn.SetReadDeadline(time.Time{})
```

don't ignore the error here